### PR TITLE
742: Add documentation to talk.xsd

### PR DESCRIPTION
**Pull request by oruam85**
This PR contain the documentation of talk.xsd

### CODE REVIEW

**Comment by alex-palevsky** [comment]
Thanks, let me find someone who can review this pull request

**Comment by alex-palevsky** [comment]
@carlosmiranda could you review this pull request

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 1]
```diff
@@ -30,43 +30,113 @@
...
+        <xs:annotation>
+            <xs:documentation source="description">Alphanumeric type value used as ID in the elements of this file.</xs:documentation>
+        </xs:annotation>
         <xs:restriction base="xs:string">
             <xs:pattern value="[0-9a-f]+"/>
         </xs:restriction>
     </xs:simpleType>
     <xs:complexType name="daemon">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification of the daemon task.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: Verb "describe" here should be "describes".

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 2]
```diff
@@ -30,43 +30,113 @@
...
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="started" type="xs:dateTime" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the started time of the daemon.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
             <xs:element name="dir" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element define the directory of the daemon.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: Verb "define" should be "defines", element here is singular.

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 3]
```diff
@@ -30,43 +30,113 @@
...
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the description of the daemon.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
         </xs:all>
         <xs:attribute name="id" type="hash" use="required"/>
     </xs:complexType>
     <xs:complexType name="shell">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification about the shell task.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: Same here - "This type *describes*..."

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 4]
```diff
@@ -30,43 +30,113 @@
...
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the password to use in the connection.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
         </xs:all>
         <xs:attribute name="id" type="hash" use="required"/>
     </xs:complexType>
     <xs:complexType name="request">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification about the request task.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: Same here, "describe" should be "describes"

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 5]
```diff
@@ -30,43 +30,113 @@
...
+        </xs:annotation>
         <xs:all>
-            <xs:element name="success" type="xs:boolean" minOccurs="0"/>
-            <xs:element name="msec" type="xs:integer" minOccurs="0"/>
-            <xs:element name="highlights" type="xs:string" minOccurs="0"/>
+            <xs:element name="success" type="xs:boolean" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the flag that indicate if the request was successfully.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: "Indicate" should be "indicates".

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 6]
```diff
@@ -30,43 +30,113 @@
...
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="msec" type="xs:integer" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the number of milliseconds that takes the request.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="highlights" type="xs:string" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the description about the request task.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: "This element has the description *of* the request task."

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 7]
```diff
@@ -74,9 +144,15 @@
                 </xs:simpleType>
             </xs:element>
             <xs:element name="args">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the list of argument of the request.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: "...has the list of *arguments*..."

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 8]
```diff
@@ -103,16 +182,41 @@
...
+            <xs:element name="github-seen" type="xs:integer" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the number of views of the issue.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
         </xs:all>
     </xs:complexType>
     <xs:complexType name="archive">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification of the archive task.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: "describe" -> "describes"

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 9]
```diff
@@ -126,17 +230,47 @@
         </xs:sequence>
     </xs:complexType>
     <xs:complexType name="ec2">
+        <xs:annotation>
+            <xs:documentation source="description">This type describe the specification about the EC2.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: Here also, verb should be "describes".

File: src/main/resources/com/rultor/spi/talk.xsd [hunk 10]
```diff
@@ -30,43 +30,113 @@
...
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="port" type="xs:integer">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the port to use in the connection.</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="login" type="xs:string">
+                <xs:annotation>
+                    <xs:documentation source="description">This element has the user to use in the connection.</xs:documentation>
```
Comment (carlosmiranda) [inline src/main/resources/com/rultor/spi/talk.xsd]: I think it's better to restate it as "This element has the username of the connection"

**Comment by carlosmiranda** [comment]
@oruam85 Some comments above, mostly English grammar. :)

**Comment by oruam85** [comment]
@carlosmiranda Please review again.

**Comment by carlosmiranda** [comment]
@rultor Merge please.

**Comment by yegor256** [comment]
@oruam85 we don't have a static analyzer for XML files, but this one is obviously in trouble - lines are very long. You see how difficult it is to review the file in Github? Make them 80 chars width, will be much more elegant and easier to read

**Comment by oruam85** [comment]
@yegor256 XSD changed

**Comment by yegor256** [comment]
@rultor merge this

**Comment by alex-palevsky** [comment]
@carlosmiranda **16 mins** added to your account (payment number `50619332`), many thanks for your contribution!; +16 to your rating, your total score is [+2167](http://www.netbout.com/b/35023?open=rating)

**Comment by alex-palevsky** [comment]
@rultor deploy

**Pull request merged**